Add CurseForge modpack export

Trial97 opened the pull request:
<!--
Hey there! Thanks for your contribution.

Please make sure that your commits are signed off first.
If you don't know how that works, check out our contribution guidelines: https://github.com/PrismLauncher/PrismLauncher/blob/develop/CONTRIBUTING.md#signing-your-work
If you already created your commits, you can run `git rebase --signoff develop` to retroactively sign-off all your commits and `git push --force` to override what you have pushed already.

Note that signing and signing-off are two different things!
-->
fixes #1291 by adding curseforge export option

Trial97 committed:
Added ExportModsToStringTask

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Use slug for url

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Added curseforge export

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Updated slug for url

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
finished up the curesforge export

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
changed map tipe

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

DioEgizio commented:
This implementation has a few issues:
- ![image](https://github.com/PrismLauncher/PrismLauncher/assets/83089242/6dae2644-0d19-4025-921f-d162766d6e18) this lol
- it tries to use quilt on the CF format but that's not supported and the only way to get quilt on CF packs is through forge + jumpquilt
- it doesn't search again the mods with modrinth metadata and just uses the ones with CF metadata and adds the others to overrides
- I'm pretty sure ProjectID isn't necessary, but probably it shouldn't be 0 if not specified. Edit: I actually can't find it even on CF packs, idk where you got this from
- I'm pretty sure modlist.html isn't necessary but I don't think CF will like modpacks with modrinth links in them lol

TheKodeToad commented:
I guess reusing the dialog is clever, but you will need to fix some things, yeah ^^

DioEgizio commented:
Extra info: for the request for searching on cf the mods you need to exclude the mods with isAvailable false since those are archived and can't be used on cf. Great way to test this is through 1.17+ Sodium

DioEgizio commented:
found another issue: it puts required:true (and  not false) even when the mod is disabled

DioEgizio commented:
i tried exporting from the cf app and:
there's no ProjectID (outside files ofc)
author and version can be both "", so you can remove the default 1.0.0 on cf (but not on mr) and allow making packs without it

Trial97 committed:
Merge branch 'develop' of https://github.com/PrismLauncher/PrismLauncher into curse

Trial97 committed:
Added curseforge search

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Merge branch 'develop' of https://github.com/PrismLauncher/PrismLauncher into curse

Trial97 committed:
Fixed merge

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Fixed hash checking

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Merge branch 'develop' of https://github.com/PrismLauncher/PrismLauncher into curse

Trial97 commented:
Here is a small update regarding comments:
- the dialog title was changed
- it searches for files(only mods) if they do not have already metadata or if they are not from curse
- projected is now optional; if not completed it is committed from manifest.json
- the require field takes into consideration if the mod is disabled
- all the other fields from the dialog are empty by default

Not yet fixed but with questions:
- if modlist.html is not required for the current format can the fact that it points to mods outside curse be ignored? 
- the quilt stuff: from what I see it's planned to be added in the future so can that also be ignored or should I try to implement a workaround?

DioEgizio commented:
Knowing CF it will take ages for them to do it. Tbh you could even just add an error message when using Quilt, I don't think many people using quilt would use cf, perhaps you could make a separate issue for it/it could be in a different PR
While the modlist.html is not required (I think, at least afaik) I don't think CF would like having modrinth urls in modpacks of their platform
Also I think you could just remove projectID as i don't think it has any use tbh (the official cf app doesn't even add it), and most people won't know what it means 

Also are you checking if isAvailable is true? We can't have archived mods in cf packs

Trial97 commented:
ok, will add an error for quilt.
I removed the projectID, but you can still populate it from the dialog
Let's leave the users to do their mistakes regarding adding modrinth urls in mod packs
Regarding isAvailable = true I found it in the documentation of curseforge.

DioEgizio commented:
I mean it's not like they can really control the links that are in the packs. I think you could even just exclude the modrinth mods in the modlist.html (and just generate only from the CF mods after searching them)

Trial97 committed:
Added some warnings

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 commented:
This should also address isAvailable stuff and also the control over if the modlist is generated or not(with obviously curseforge links)

DioEgizio commented on launcher/ui/MainWindow.cpp:
```diff
@@ -1416,6 +1417,23 @@ void MainWindow::on_actionExportInstanceMrPack_triggered()
+        if (instance) {
+            if (instance->getPackProfile()->getComponent("org.quiltmc.quilt-loader")) {
+                QMessageBox msgBox;
+                msgBox.setText(tr("Quilt is not yet supported by curseforge."));
```
```suggestion
                msgBox.setText(tr("Quilt is not currently supported by CurseForge modpacks."));
```

DioEgizio reviewed (commented):
also i don't think an option for not generating the modlist makes much sense, as most users won't know what it does. I think we can just always generate it but with these changes (only cf links).
I'm also not sure of the point of the projectID option as I don't think it's even part of the cf modpack format
Apart from these and the fact that windows build is failing LGTM

Trial97 committed:
Update launcher/ui/MainWindow.cpp

Co-authored-by: DioEgizio <83089242+DioEgizio@users.noreply.github.com>
Signed-off-by: Alexandru Ionut Tripon <alexandru.tripon97@gmail.com>

Trial97 committed:
Merge branch 'develop' of https://github.com/PrismLauncher/PrismLauncher into curse

Trial97 committed:
Merge branch 'curse' of github.com:Trial97/PrismLauncher into curse

DioEgizio commented on launcher/modplatform/flame/FlamePackExportTask.cpp:
```diff
@@ -0,0 +1,430 @@
+#include "modplatform/helpers/HashUtils.h"
+#include "tasks/Task.h"
+
+const QString FlamePackExportTask::TEMPLATE = "<li><a href={url}>{name}({authors})</a></li>";
```
The CF Launcher does it as [name] (by [author])
So it should be 
```suggestion
const QString FlamePackExportTask::TEMPLATE = "<li><a href={url}>{name} (by {authors})</a></li>";
```

Trial97 committed:
Removed modlist checkbox

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Update launcher/modplatform/flame/FlamePackExportTask.cpp

Co-authored-by: DioEgizio <83089242+DioEgizio@users.noreply.github.com>
Signed-off-by: Alexandru Ionut Tripon <alexandru.tripon97@gmail.com>

DioEgizio commented:
Another thing that could be done is adding rpacks to the json. Basically you only have to do the search on rpacks too and if you find them adding them to the manifest.json like you would for mods

TheKodeToad commented:
> also i don't think an option for not generating the modlist makes much sense, as most users won't know what it does. I think we can just always generate it but with these changes (only cf links). I'm also not sure of the point of the projectID option as I don't think it's even part of the cf modpack format Apart from these and the fact that windows build is failing LGTM

Prism Launcher is a pretty advanced application though, right? lol
Tooltips are useful, and I'm pretty sure most users will know what HTML is!
(so it could be "Generate a HTML file containing a mod list inside the modpack.")

TheKodeToad commented on launcher/modplatform/ModIndex.cpp:
```diff
@@ -70,11 +70,18 @@ auto ProviderCapabilities::hash(ResourceProvider p, QIODevice* device, QString t
     return { hash.result().toHex() };
 }
 
+QString getMetaURL(ResourceProvider provider, QString slug)
+{
+    return ((provider == ModPlatform::ResourceProvider::FLAME) ? "https://www.curseforge.com/minecraft/mc-mods/"
+                                                               : "https://modrinth.com/mod/") +
+           slug.remove(".pw.toml");
+}
```
Can you use the `slug` field from `metadata()`?
Besides that, using `remove` feels a bit fragile!

Trial97 replied:
nope, I created this function because I did not have a metadata object: when exporting I do not create the Metadata but have access to the provider and slug.
The slug for some reason when comes from files is having that suffix so I just need to remove it in order to obtain the slug used by CurseForge. Do you want to check id it's a suffix with `endsWith` and remove it only then?

TheKodeToad replied:
Oh, you are passing in the slug field.
I expected it to not have the suffix, i guess it's just a misleading field name

I guess this works

TheKodeToad commented on launcher/modplatform/flame/FlamePackExportTask.cpp:
```diff
@@ -0,0 +1,422 @@
+void FlamePackExportTask::collectHashes()
+{
+    setAbortable(true);
+    setStatus(tr("Find file hashes..."));
```
I think this is best to keep the same as MR tbh

Trial97 replied:
What should be the same: the message(I do not see any message in MR) or the fact that the user can abort the task(I tested and works)?

TheKodeToad replied:
remove the message or add it to modrinth

TheKodeToad commented on launcher/modplatform/flame/FlamePackExportTask.cpp:
```diff
@@ -0,0 +1,422 @@
+
+void FlamePackExportTask::makeApiRequest()
+{
+    setStatus(tr("Find versions for hashes..."));
```
Same here...

Trial97 replied:
On MR there is no message for this

TheKodeToad commented on launcher/ui/dialogs/ExportMrPackDialog.cpp:
This needs a different name

TheKodeToad commented on launcher/ui/dialogs/ExportMrPackDialog.cpp:
```diff
@@ -91,26 +101,36 @@ void ExportMrPackDialog::done(int result)
+                                                  nullptr);
+        else
+            output = QFileDialog::getSaveFileName(this, tr("Export %1").arg(ui->name->text()),
+                                                  FS::PathCombine(QDir::homePath(), filename + ".zip"), "Curseforge pack (*.zip)", nullptr);
```
CurseForge

DioEgizio commented:
> > also i don't think an option for not generating the modlist makes much sense, as most users won't know what it does. I think we can just always generate it but with these changes (only cf links). I'm also not sure of the point of the projectID option as I don't think it's even part of the cf modpack format Apart from these and the fact that windows build is failing LGTM
> 
> Prism Launcher is a pretty advanced application though, right? lol Tooltips are useful, and I'm pretty sure most users will know what HTML is! (so it could be "Generate a HTML file containing a mod list inside the modpack.")

I mean that for the end user it wouldn't change anything. It's literally an html that's technically part of CF packs but it's basically useless. And there's no reason to *not* have it, my only concern was modrinth links which could have made cf mad but now it's fixed

TheKodeToad commented:
In the past I found it handy because I couldn't get the cf app on linux :p

Trial97 committed:
Fixed windows build

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Merge branch 'curse' of github.com:Trial97/PrismLauncher into curse

TheKodeToad commented on launcher/modplatform/flame/FlamePackExportTask.h:
```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (c) 2023 Trial97 <alexandru.tripon97@gmail.com>
```
the file looks like it was pasted from ModrinthPackExportTask

Trial97 replied:
Yeah it was pasted from there(only changed the Copyright line. What should I change?

TheKodeToad replied:
I thought you probably need to keep the copyright line. i don't personally mind though!

Trial97 replied:
If you want I can add you there :smile_cat:

TheKodeToad replied:
Point is.. I think you may need to do this

Trial97 replied:
sure done

TheKodeToad commented on launcher/modplatform/helpers/ExportModsToStringTask.cpp:
I don't think this is used.

TheKodeToad commented on launcher/modplatform/flame/FlamePackExportTask.cpp:
```diff
@@ -0,0 +1,422 @@
+
+void FlamePackExportTask::getProjectsInfo()
+{
+    setStatus(tr("Find project info from curseforge..."));
```
CurseForge

TheKodeToad reviewed (changes requested):
Messy review (sorry)

Trial97 committed:
Added resource packs to export

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Made sure CurseForge string is corect

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Removed unused files

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Renamed ExportMrPackDialog to ExportPackDialog

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Added condition for modlist

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Updated Modrinth esport messages

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Updated copyright headers

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

DioEgizio commented:
<img width="474" alt="IMAGE" src="https://github.com/PrismLauncher/PrismLauncher/assets/83089242/a129a053-1cc3-42bc-b677-5014ff6641b8">

huh i think i broke smth in https://github.com/PrismLauncher/PrismLauncher/pull/1249/commits/a325d814e242ee450141088513d9cd282d79141f

Trial97 committed:
Merge branch 'develop' of https://github.com/PrismLauncher/PrismLauncher into curse

Trial97 committed:
Updated authors string in modlist

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 commented:
Is not because of the mentioned commit. 
The mods have metadata associated with them and are from CurseForge so the authors are from the jar and may not be populated. 
So I added more code to omit that section if the authors are not present.
Also, I'm thinking to extend a little the pacwiz format to store extra information regarding the mods (e.g. authors and dependencies and so on) 
But that is far from the scope of this PR.

Trial97 committed:
Updated url function

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

DioEgizio commented:
now it works but i don't understand why it's not populating it even if it reads the metadata
<img width="439" alt="image" src="https://github.com/PrismLauncher/PrismLauncher/assets/83089242/9e07a6e0-8340-484a-8b6d-a1aff8f16331">
It worked some commits before



<img width="624" alt="?" src="https://github.com/PrismLauncher/PrismLauncher/assets/83089242/67fe6f10-51be-4436-a9cd-7eef4d047750">

also i don't get what is this for. I've never seen it in any modpack and not even the cf app adds any of that. Can it just be removed?

Trial97 committed:
removed projectID

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 commented:
Removed projectID for good
~~Also found the problem with authors(is related to the `mcInstance->loaderModList()->update();` the signal received from there doesn't really triggers as I want~~
Fixed authors

Trial97 committed:
fixed authors

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

DioEgizio commented:
Two last things:
- exporting has to be disabled on snapshots as cf doesn't support it
- on vanilla technically cf supports it but we add "modLoaders":[{"primary":true}]  when there's no modloader when the cf launcher just does "modLoaders":[]. It's not very useful to export a vanilla instance but still it shouldn't export a non working zip

Trial97 committed:
Added error message when exporting snapshots with curseforge

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

TheKodeToad commented on launcher/modplatform/flame/FlamePackExportTask.cpp:
```diff
@@ -0,0 +1,440 @@
+
+    for (const QFileInfo& file : files) {
+        const QString relative = gameRoot.relativeFilePath(file.absoluteFilePath());
+        if (!relative.endsWith(".zip") || !relative.startsWith("resourcepacks/"))
```
:P maybe you could try to mimic mrpack more
this logic doesn't really make sense

Trial97 replied:
this should be quite similar to mrpack: here I'm interested only in resourcepacks so I look only at that folder and that they have a specific extension.
The mrpack does the same but with multiple resource types here:(https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/modplatform/modrinth/ModrinthPackExportTask.cpp#L97-L108)
As I know CurseForge only has mods and resource packs in a mod pack(correct me if I'm wrong), so I chose to parse the mods first(using all the mods list) and then parse the resource packs separately by parsing selected files

Trial97 replied:
in the end did make it more similar

TheKodeToad replied:
> this should be quite similar to mrpack: here I'm interested only in resourcepacks so I look only at that folder and that they have a specific extension. The mrpack does the same but with multiple resource types here:(https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/modplatform/modrinth/ModrinthPackExportTask.cpp#L97-L108) As I know CurseForge only has mods and resource packs in a mod pack(correct me if I'm wrong), so I chose to parse the mods first(using all the mods list) and then parse the resource packs separately by parsing selected files

The problem is it's checking for things which are either .zip or (rather than and) in resourcepacks

DioEgizio replied:
zips can also be mods

TheKodeToad replied:
there's no condition for jar

TheKodeToad replied:
I feel like your previous code was better than in mrpack now, however, it should be && rather than ||

Trial97 replied:
sorry I did forget old code there now it should be ok

Trial97 committed:
Made it more similar to mrpack export

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

DioEgizio commented on launcher/modplatform/flame/FlamePackExportTask.cpp:
```diff
@@ -0,0 +1,471 @@
+
+const QString FlamePackExportTask::TEMPLATE = "<li><a href={url}>{name}{authors}</a></li>\n";
+const QStringList FlamePackExportTask::PREFIXES({ "mods/", "resourcepacks/" });
+const QStringList FlamePackExportTask::FILE_EXTENSIONS({ "jar", "zip" });
```
Kode isn't this the jar condition?

TheKodeToad replied:
yeah, and zip's there :p

Trial97 committed:
Removed dupliacte code

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 commented:
Ok let me reformulate:
- at this ifs: https://github.com/PrismLauncher/PrismLauncher/pull/1249/files#diff-4b8c3382fb8302697895f9a6a75bdd2adc46377afe99c8924a52ace40e889eafR119-R124 I check the prefixes and sufixes
- here: https://github.com/PrismLauncher/PrismLauncher/pull/1249/files#diff-4b8c3382fb8302697895f9a6a75bdd2adc46377afe99c8924a52ace40e889eafR126-R139 I check only for resource packs
- and finally if is not a resourcepack I check if is a mod here: https://github.com/PrismLauncher/PrismLauncher/pull/1249/files#diff-4b8c3382fb8302697895f9a6a75bdd2adc46377afe99c8924a52ace40e889eafR141-R165

If the file doesn't match any of these categories the file should not be hashed and send to the overrides folder 
I did go with the mrpack export in order to be able to select what mods enter in the pack(previously all mods were included)

TheKodeToad committed:
Simplify

Signed-off-by: TheKodeToad <TheKodeToad@proton.me>

TheKodeToad commented:
I removed some stuff which shouldn't be necessary. i hope that's okay :p

DioEgizio commented:
mods can be .zip
See https://github.com/PrismLauncher/PrismLauncher/issues/349

TheKodeToad commented:
Why wouldn't they be picked up by this? Are they not in `allMods()`?

Trial97 commented:
> mods can be .zip See #349

For mods I do not check to be jars only I check for the file to be in allMods list.

TheKodeToad committed:
Remove metaurl function

Signed-off-by: TheKodeToad <TheKodeToad@proton.me>

DioEgizio commented:
Ooh then it's fine

TheKodeToad commented:
Thank you! I didn't have the courage to make this feature

i would approve but the progress handling seems to be a bit weird atm...

Trial97 commented:
> i would approve but the progress handling seems to be a bit weird atm...

Can you point me to a better example of how progress handling should be done? Or can you be more exact with what's weird?

TheKodeToad commented:
The maximum progress is changing.

Trial97 committed:
Removed some setProgress calls

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 committed:
Merge branch 'curse' of github.com:Trial97/PrismLauncher into curse

Trial97 commented:
Removed some setProgress stuff hope this fixes the weird progress stuff.

TheKodeToad commented:
It's still changing though. I'd suggest just setting it to to files' length

Trial97 committed:
Complicated a little task progress

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

DioEgizio commented:
> It's still changing though. I'd suggest just setting it to to files' length

I suppose that's fixed in the latest commit?

Trial97 commented:
It ~~should be~~ is fixed(~~it moves too fast for me~~). But instead of using just one progress bar, I use two: one for the current step and one for the progress of the step

TheKodeToad commented:
There seems to be a conflict; i'll try to review once that is fixed

![image](https://github.com/PrismLauncher/PrismLauncher/assets/57493648/113ea963-e28d-48de-8004-d5bfe6a9421f)

Hopefully this can be fixed too :/

Trial97 committed:
Merge branch 'develop' of https://github.com/PrismLauncher/PrismLauncher into curse

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

TheKodeToad commented on launcher/modplatform/flame/FlamePackExportTask.cpp:
```diff
@@ -0,0 +1,473 @@
+            auto hash_task = Hashing::createFlameHasher(file.absoluteFilePath());
+            connect(hash_task.get(), &Hashing::Hasher::resultsReady, [this, relative, file](QString hash) {
+                if (m_state == Task::State::Running) {
+                    pendingHashes.insert(hash, { relative, file.absoluteFilePath(), relative.endsWith(".zip") });
```
I believe this will cause issues since the .disabled extension isn't being removed
Same with mods!

Trial97 replied:
I'm not sure I understand. ?
If it has the extension then that means it's not required( CurseForge has a field for this in the manifest file:`required`).
If I remove the extension then the hash task doesn't find the file to get the hash, if let it there then I can not really match it after I find the hash.
Did you have trouble with the fact that a resource was disabled on export?
Also, the Modrinth version doesn't care about the extension other than when filtering files for hashing.

TheKodeToad replied:
Yeah, so doesn't it mean you're indicating it's not required in two places? therefore it will become mod.jar.disabled.disabled

But I guess it would create complications to rename it.. hmm

Trial97 replied:
nop I do not modify the filename in any way

Trial97 replied:
The CurseForge manifest file doesn't track where that file is located so no name needed there

TheKodeToad replied:
Oh ok :woman_facepalming: 

Sorry about that. This will be needed with Modrinth export though!

TheKodeToad commented on launcher/modplatform/flame/FlamePackExportTask.cpp:
```diff
@@ -0,0 +1,473 @@
+void FlamePackExportTask::collectHashes()
+{
+    setAbortable(true);
+    setStatus(tr("Find file hashes..."));
```
Should be "Finding file hashes..." tbh

TheKodeToad commented on launcher/modplatform/flame/FlamePackExportTask.cpp:
```diff
@@ -0,0 +1,473 @@
+        return;
+    }
+
+    setStatus(tr("Find versions for hashes..."));
```
`Finding versions for hashes...`

TheKodeToad commented on launcher/modplatform/flame/FlamePackExportTask.cpp:
```diff
@@ -0,0 +1,473 @@
+
+void FlamePackExportTask::getProjectsInfo()
+{
+    setStatus(tr("Find project info from CurseForge..."));
```
`Finding project info from CurseForge...`

TheKodeToad commented on launcher/modplatform/flame/FlamePackExportTask.cpp:
```diff
@@ -0,0 +1,473 @@
+            hashing_task->addTask(hash_task);
+        }
+    }
+    auto step_progress = std::make_shared<TaskStepProgress>();
```
Confusing - looks like the stepProgress method

Trial97 replied:
I know but this is used with the same name in multiple places so I think I will keep this with this name.

Trial97 replied:
Renamed

TheKodeToad commented on launcher/modplatform/flame/FlamePackExportTask.cpp:
```diff
@@ -0,0 +1,473 @@
+    setStatus(tr("Find project info from CurseForge..."));
+    setProgress(3, 5);
+    QList<QString> addonIds;
+    for (auto resolved : resolvedFiles) {
```
Apparently this should be a reference (thanks ide)
i-i knew that

TheKodeToad reviewed (changes requested):
I don't think ExportPackDialog.ui should have any references to Modrinth.


I also think that either snake case or camel case should be used, not both.

Trial97 committed:
fixed typos

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 commented:
> I also think that either snake case or camel case should be used, not both.

I'm ok to change the case but I'm not sure what's the best here.
Does the project have a guideline regarding how to name variables?
If I go choosing one I need to make that change for all the variables in the modified files?
Also, I find this confusing there are cases where the variables are a combination of snake and camel case(e.g. `m_relativePath`)
** The reason why I mix them is that I find that camel case being more easy to read.

TheKodeToad commented:
Some local variables look like `var_name` and others look like `varName`. That's what I meant.

Trial97 committed:
resolved local vaiables names

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 commented:
Ok. I updated the local variables

TheKodeToad reviewed (approved):
Sorry, I've had trouble focusing :P

Looks good, it doesn't look like the HTML is escaped properly though!
Unlikely this will cause issues I guess...

Trial97 committed:
fixed template

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

TheKodeToad commented:
Oh, I was talking about the inner HTML being set instead of inner text but it's probably not trivial to fix properly

Trial97 committed:
escaped modlist inner html

Signed-off-by: Trial97 <alexandru.tripon97@gmail.com>

Trial97 commented:
Ohhh, that's what you meant. Well this should solve that
